File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/published/bun/projects/l3-component-config-primitives/primitiveComponent.ts
import * as pulumi from "@pulumi/pulumi";
import * as primitive from "@pulumi/primitive";

interface PrimitiveComponentArgs {
    boolean: pulumi.Input<boolean>,
    float: pulumi.Input<number>,
    integer: pulumi.Input<number>,
    string: pulumi.Input<string>,
}

export class PrimitiveComponent extends pulumi.ComponentResource {
    constructor(name: string, args: PrimitiveComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:PrimitiveComponent", name, args, opts);
        const res = new primitive.Resource(`${name}-res`, {
            boolean: args.boolean,
            float: args.float,
            integer: args.integer,
            string: args.string,
            numberArray: [
                -1,
                0,
                1,
            ],
            booleanMap: {
                t: true,
                f: false,
            },
        }, {
            parent: this,
        });

        this.registerOutputs();
    }
}

File: sdk/python/cmd/pulumi-language-python/testdata/setuppy/published/projects/l3-component-config-objects/primitiveComponent.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import builtins as _builtins
import pulumi_primitive as primitive

class PrimitiveComponentArgs(TypedDict):
    numberArray: Input[list[_builtins.float]]
    booleanMap: Input[Dict[_builtins.str, _builtins.bool]]

class PrimitiveComponent(pulumi.ComponentResource):
    def __init__(self, name: str, args: PrimitiveComponentArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:PrimitiveComponent", name, args, opts)

        res = primitive.Resource(f"{name}-res",
            boolean=True,
            float=3.5,
            integer=3,
            string="plain",
            number_array=args["numberArray"],
            boolean_map=args["booleanMap"],
            opts = pulumi.ResourceOptions(parent=self))

        self.register_outputs({})
